File: src/frontend/my-RacksStand/src/app/stores/shared/store-model.ts
export class StoreModel {
        public  Id :string ;
        public  Name :string ;
        public  Description:string ;
        public  CompanyId:string ; 
        public City: string;
        public  Country: number;
        public Email: string;
        public  Fax: string;
        public  Phone: string;
        public  State: string;
        public  Street: string;
        public   ZipCode: string;
        public  Status: number;
        public  CreatedOn: Date;
        public  CreatedBy: string;
        public UpdatedOn: Date;
        public SecondaryStatus:number;
        public Index:number;
        public UpdatedBy: string;
        public Rooms:RoomModel[];
}
export class RoomModel {
        public  Id :string ;
        public  Name :string ;
        public  Description:string ;
        public  StoreId:string ; 
        public  Status: number;
        public SecondaryStatus:number;
        public  CompanyId:string ; 
        public Index:number;
        public  CreatedOn: Date;
        public  CreatedBy: string;
        public UpdatedOn: Date;
        public UpdatedBy: string;
        public Racks:RackModel[];
}
export class RackModel {
        public  Id :string ;
        public  Name :string ;
        public  Description:string ;
        public  RoomId:string ; 
        public Rows:number;
        public Columns:number;
        public  Status: number;
        public BoxCapacity:number;
        public SecondaryStatus:number;
        public  CompanyId:string ; 
        public Index:number;
        public  CreatedOn: Date;
        public  CreatedBy: string;
        public UpdatedOn: Date;
        public UpdatedBy: string;
        public Boxes:RackBoxModel[];
}
export class RackBoxModel {
        public  Id :string ;
        public  Name :string ;
        public  Description:string ;
        public  RackId:string ;
        public CurrentSize:number; 
        public SecondaryStatus:number;
        public  CompanyId:string ; 
        public Index:number;
        public Row:number;
        public Column:number;
  }

File: src/backend/BusinessLogic/Models/Core/StoreModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Models.Core
{
    [Serializable]
    public class StoreModel
    {
        public StoreModel()
        {
            this.Rooms = new List<RoomModel>();
        }
        [Required]
        [MaxLength(32)]
        public string Id { get; set; }
        [Required]
        [MaxLength(100)]
        public string Name { get; set; }
        [Required]
        [MaxLength(150)]
        public string Street { get; set; }
        [MaxLength(250)]
        public string Description { get; set; }
        public string CompanyId { get; set; }
        [Required]
        [Range(0, 255)]
        public byte Country { get; set; }
        [Required]
        [MaxLength(50)]
        public string State { get; set; }
        [Required]
        [MaxLength(50)]
        public string City { get; set; }
        [EmailAddress]
        public string Email { get; set; }
        public string Fax { get; set; }
        [Phone]
        public string Phone { get; set; }
        [Required]
        [MaxLength(20)]
        public string ZipCode { get; set; }
        public int? Index { get; set; }
        public byte SecondaryStatus { get; set; }
        public byte Status { get; set; }
        public string UpdatedBy { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public ICollection<RoomModel> Rooms { get; set; }
    }
}
